src/lexer.coffee

```diff
@@ -900,6 +890,19 @@ exports.Lexer = class Lexer
           addTokenData token, {heregex} if heregex
           token[0] = 'STRING'
           token[1] = '"' + converted + '"'
+          if tokens.length is 1 and quote?
```

Comment:
So here we're sort of "re-adding" the code that was removed above - only in the case that this is a non-interpolated string do we want the token's location data to include the surrounding quotes/delimiters (since at that point the token represents an entire string literal rather than just a sub-piece of an interpolated string)

Reply:
Does this match what Babel would produce for source maps for strings?

Reply:
I don't know enough about source maps to fully answer this (though I'd be curious to know more if you want to explain). I'd dug as deep as to see that yes this is more similar to the tokens generated by the Babel parser:

For a simple string literal eg `"a"`, it represents that as a single token of type `string`

For an interpolated string (template literal) eg `` `a${b}` ``, its token stream looks like `` ` ``, `a`, `${`, `b`, `}`, [empty text], `` ` ``, where `a` and the empty text are a `template` token type

Reply:
Sorry I just meant does it match in terms of the line/column numbers. Like is Babel inclusive or exclusive? Since you’re changing CoffeeScript’s source maps for strings from inclusive to exclusive, I’m wondering which way Babel is.

Reply:
I believe the source mapping (based on location data) should stay consistent - basically the only thing that should be changing is the location data for a token that represents a "text chunk" of an interpolated string. The location data for a string literal (both its token and its resulting `StringLiteral` node class instance) and for an interpolated string (`StringWithInterpolations` instance) should be the same as they were previously

> Since you’re changing CoffeeScript’s source maps for strings from inclusive to exclusive

So if I'm understanding correctly, this change in behavior isn't about inclusive/exclusive, it's that location data for leading/trailing "text chunk" tokens in an interpolated string will no longer include the preceding/following quote in its location data (which in the existing behavior overlapped their location data with the adjacent `STRING_START`/`STRING_END` tokens, which are what actually represent the start/end quotes of the interpolated string)

Eg in `"a#{b}c"`, the location data range for the `a` token will now be `[1, 2]` instead of `[0, 2]` and for the `c` token will be `[6, 7]` instead of `[6, 8]`

So I think this won't have consequences for source maps. But if you think it deserves deeper investigation we could compare sourcemaps before and after this PR, I'm just not familiar with how to do eg a visual diff of the mappings created by sourcemap files

Reply:
I’m just a) nervous that this might be a breaking change to the non-AST side of things, and b) wanting to make sure that if our `a` in your example is `[1, 2]`, than the Babel source map for `` `a${b}c` `` would have its `a` also as `[1, 2]`. Like if we’re going to change at all, we should align with Babel here.

Reply:
The mapping of `a` to `[1, 2]` is definitely more correct (and matches Babel's token location data) as that's the actual location of that token

As far as sourcemaps, I did a visual diff of the sourcemap for `"a#{b}c"` pre- and post-this PR using [this tool](https://sokra.github.io/source-map-visualization/)

The only difference in the mapping is that `a` (in the output `` `a${b}c`;``) now gets mapped to `a` in the source (ie line 1, column 1) whereas previously it was getting mapped to the leading `"` in the source (line 1, column 0)

So it's a slightly more accurate sourcemap that's being generated